test/utils/objects.go

```diff
@@ -85,6 +88,36 @@ func PrepareTestObjects(ctx context.Context, c client.Client, namespacePrefix st
 	return o
 }
 
+// CleanupTestObjects take care to remove the finalizers of the secret and configmaps
+func CleanupTestObjects(ctx context.Context, o *TestObjects) {
```

Comment:
I would suggest to add a call to this function to the existing cleanup handler before line 46. 
This way, you don't have to explicitly call the function in every `AfterEach`. WDYT?

Reply:
Done. 
I had done it this way but the finalizers were not removed. It turned out to be completely different problem, but I forgot to move it back to the already existing clean up handle.